pkgs/misc/emulators/pcsx2/default.nix

```diff
@@ -1,57 +1,80 @@
-{ alsaLib, cmake, fetchFromGitHub, glib, gtk2, gettext, libaio, libpng
-, makeWrapper, perl, pkgconfig, portaudio, SDL2, soundtouch, stdenv
-, wxGTK30, zlib }:
+
+
+{ alsaLib, cmake, fetchFromGitHub, glib, gettext, harfbuzz, lib, libaio
+, libpng, libpcap, libxml2, makeWrapper, perl, pkgconfig, portaudio
+, SDL2, soundtouch, stdenv, udev, wxGTK, zlib
+
+, gtk2Support ? true, gtk2
+, gtk3Support ? false, gtk3
+}:
+
+assert gtk2Support -> !gtk3Support;
+assert gtk2Support || gtk3Support;
 
 stdenv.mkDerivation rec {
   pname = "pcsx2";
-  version = "1.4.0";
+  version = "unstable-2019-11-26";
 
   src = fetchFromGitHub {
     owner = "PCSX2";
     repo = "pcsx2";
-    rev = "v${version}";
-    sha256 = "0s7mxq2cgzwjfsq0vhpz6ljk7wr725nxg48128iyirf85585l691";
+    rev = "2827394f3cbc6a2ea3219583e12412bfa34852ab";
+    sha256 = "16ixlhhlfmwy0ac2yirwj3bv2s83bhxap4z4ha7npagahl3n8j9f";
   };
 
   postPatch = "sed '1i#include \"x86intrin.h\"' -i common/src/x86emitter/cpudetect.cpp";
 
-  configurePhase = ''
-    mkdir -p build
-    cd build
-
-    cmake \
-      -DBIN_DIR="$out/bin" \
-      -DCMAKE_BUILD_PO=TRUE \
-      -DCMAKE_BUILD_TYPE=Release \
-      -DCMAKE_INSTALL_PREFIX="$out" \
-      -DDISABLE_ADVANCE_SIMD=TRUE \
-      -DDISABLE_PCSX2_WRAPPER=TRUE \
-      -DDOC_DIR="$out/share/doc/pcsx2" \
-      -DGAMEINDEX_DIR="$out/share/pcsx2" \
-      -DGLSL_SHADER_DIR="$out/share/pcsx2" \
-      -DGTK2_GLIBCONFIG_INCLUDE_DIR='${glib.out}/lib/glib-2.0/include' \
-      -DGTK2_GDKCONFIG_INCLUDE_DIR='${gtk2.out}/lib/gtk-2.0/include' \
-      -DGTK2_INCLUDE_DIRS='${gtk2.dev}/include/gtk-2.0' \
-      -DPACKAGE_MODE=TRUE \
-      -DPLUGIN_DIR="$out/lib/pcsx2" \
-      -DREBUILD_SHADER=TRUE \
-      -DXDG_STD=TRUE \
-      ..
-  '';
+  cmakeFlags = [
+    "-DBIN_DIR=${placeholder "out"}/bin"
+    "-DCMAKE_BUILD_PO=TRUE"
+    "-DCMAKE_BUILD_TYPE=Release"
```

Comment:
This is default value with cmake setup hook.

Reply:
Thanks!